### ISSUE TITLE
RabbitMQ reactive messaging health check cannot be disabled

### ISSUE THREAD

**Issue by jdussouillez** [quarkusio/quarkus#31977]
### Describe the bug

RabbitMQ reactive messaging health check cannot be disabled using [`health-enabled`](https://quarkus.io/guides/rabbitmq-reference#health-reporting). It seems the option has no effect and health check are still enabled if I look in the Dev UI.

Configuration :
```
# ...
mp.messaging.outgoing.quote-requests.health-enabled=false
mp.messaging.outgoing.quote-requests.health-readiness-enabled=false

# ...
mp.messaging.incoming.quotes.health-enabled=false
mp.messaging.incoming.quotes.health-readiness-enabled=false
```

![health_enabled](https://user-images.githubusercontent.com/5428355/226338544-7acc0d7f-6a1b-48f1-a7e8-de30d8a0e455.png)

These checks are disabled if I use [`[quarkus.health.extensions.enabled](https://quarkus.io/guides/smallrye-health#quarkus-smallrye-health_quarkus.health.extensions.enabled)`](https://quarkus.io/guides/smallrye-health#quarkus-smallrye-health_quarkus.health.extensions.enabled) but I want to keep some checks on other extensions.

### Expected behavior

The following health check must be disabled:
- "SmallRye Reactive Messaging - liveness check"
- "SmallRye Reactive Messaging - readiness check"
- "SmallRye Reactive Messaging - startup check"

### Actual behavior

The heath checks are still enabled.

### How to Reproduce?

- Clone https://github.com/jdussouillez/quarkus-rabbitmq-health-check
- Run `./mvnw quarkus:dev`
- Go to http://localhost:8080/q/health-ui/

### Output of `uname -a` or `ver`

Linux 5.4.0-144-generic #161-Ubuntu SMP Fri Feb 3 14:49:04 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "17.0.6" 2023-01-17 OpenJDK Runtime Environment (build 17.0.6+10-Ubuntu-0ubuntu120.04.1) OpenJDK 64-Bit Server VM (build 17.0.6+10-Ubuntu-0ubuntu120.04.1, mixed mode, sharing)

### Quarkus version or git rev

2.16.4.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.6 (84538c9988a25aec085021c365c560670ad80f63) Maven home: ~/.m2/wrapper/dists/apache-maven-3.8.6-bin/67568434/apache-maven-3.8.6 Java version: 17.0.6, vendor: Private Build, runtime: /usr/lib/jvm/java-17-openjdk-amd64 Default locale: en_US, platform encoding: UTF-8 OS name: "linux", version: "5.4.0-144-generic", arch: "amd64", family: "unix"

**Comment by jdussouillez** [comment]
Might be related to https://github.com/quarkusio/quarkus/issues/15321

**Comment by jdussouillez** [comment]
This is still happing with Quarkus 3.0.3.Final

**Comment by cescoffier** [comment]
@ozangunalp can't we close this issue?

**Comment by ozangunalp** [comment]
Yes, this is resolved in reactive messaging 4.14.0 and is now used by Quarkus main. Per channel health checks are documented in https://smallrye.io/smallrye-reactive-messaging/latest/rabbitmq/rabbitmq-health/
